src/lib/utils/memory_resource_manager.cpp

```diff
@@ -0,0 +1,30 @@
+#include "utils/memory_resource_manager.hpp"
+
+namespace opossum {
+
+const tbb::concurrent_vector<ResourceRecord>& MemoryResourceManager::memory_resources() const {
+  return _memory_resources;
+}
+
+boost::container::pmr::memory_resource* MemoryResourceManager::get_memory_resource(
+    const OperatorType operator_type, const std::string& operator_data_structure) {
+  // Return default memory resource if tracking is disabled.
+  if (!_tracking_is_enabled) {
+    return boost::container::pmr::get_default_resource();
+  }
+
+  auto resource_ptr = new TrackingMemoryResource(); // NOLINT
+  auto tracked_resource = ResourceRecord{operator_type, operator_data_structure, resource_ptr};
```

Comment:
Pass a smart pointer rather than a raw pointer for automatic memory management here?